src/PKSim.Core/Services/SimulationParametersToBuildingBlockUpdater.cs

```diff
@@ -51,23 +67,73 @@ public ICommand UpdateParametersFromSimulationInBuildingBlock(Simulation simulat
 
          updateCommands.Add(updateTemplateParametersCommand);
 
+         //Last, see if we have some special cases to handle 
+         var synchronizeCommand = synchronizeBuildingBlocks(templateBuildingBlock, updateTemplateParametersCommand);
+         updateCommands.Add(synchronizeCommand);
+
          //now make sure that the used building block is updated with the template building block info
          updateCommands.Add(new UpdateUsedBuildingBlockInfoCommand(simulation, usedBuildingBlock, templateBuildingBlock, _executionContext).Run(_executionContext));
-
-         updateCommands.ObjectType = buildingBlockType;
-         updateCommands.BuildingBlockType = buildingBlockType;
-         updateCommands.BuildingBlockName = templateBuildingBlock.Name;
-         updateCommands.CommandType = PKSimConstants.Command.CommandTypeUpdate;
-         updateCommands.Description = PKSimConstants.Command.UpdateTemplateBuildingBlockCommandDescription(buildingBlockType, templateBuildingBlock.Name, simulation.Name);
-         _executionContext.UpdateBuildingBlockPropertiesInCommand(updateCommands, templateBuildingBlock);
-
+       
          _executionContext.PublishEvent(new BuildingBlockUpdatedEvent(templateBuildingBlock));
          return updateCommands;
       }
 
-      private IOSPSuiteCommand updateParameterValues(PathCache<IParameter> simulationParameters, PathCache<IParameter> templateParameters)
+      private ICommand synchronizeBuildingBlocks(IPKSimBuildingBlock templateBuildingBlock, IPKSimMacroCommand updateTemplateParametersCommand)
```

Comment:
Thsi is the big change. I Am inspecitng the commit command generated (updating the template). If any of them is one of our edit command, then I am also updating the expression profile associated